tests/test_multiple_inheritance.py

```diff
@@ -53,15 +53,174 @@ def __init__(self, i, j):
     assert mt.bar() == 4
 
 
-def test_multiple_inheritance_error():
-    """Inheriting from multiple C++ bases in Python is not supported"""
+def test_multiple_inheritance_python():
     from pybind11_tests import Base1, Base2
 
-    with pytest.raises(TypeError) as excinfo:
-        # noinspection PyUnusedLocal
-        class MI(Base1, Base2):
-            pass
-    assert "Can't inherit from multiple C++ classes in Python" in str(excinfo.value)
+    class MI1(Base1, Base2):
+        def __init__(self, i, j):
+            Base1.__init__(self, i)
+            Base2.__init__(self, j)
```

Comment:
Currently nothing is raised--the space is allocated, but the constructor isn't invoked so you get UB.  I'm not sure what we should do about it, or how we can deal with it.

This isn't actually specific to multiple inheritance, though: the same thing happens if you do:

```Python
class PyDerived(CppBase):
    def __init__(self):
        pass

a = PyDerived()
a.some_cpp_method()
```